pkg/table-router/router.go

```diff
@@ -23,10 +24,59 @@ import (
 // TableRule is a rule to route schema/table to target schema/table
 // pattern format refers 'pkg/table-rule-selector'
 type TableRule struct {
-	SchemaPattern string `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
-	TablePattern  string `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
-	TargetSchema  string `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
-	TargetTable   string `json:"target-table" toml:"target-table" yaml:"target-table"`
+	SchemaPattern string     `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
+	TablePattern  string     `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
+	TargetSchema  string     `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
+	TargetTable   string     `json:"target-table" toml:"target-table" yaml:"target-table"`
+	Extractors    Extractors `json:",inline" toml:",inline" yaml:",inline"`
+}
+
+// Extractors is a rule that extracts certain values into a column
+type Extractors struct {
```

Comment:
seems if we use `omitempty`, we can keep compatibility if not enable them.

Reply:
Hi @yufan022 don't forget this comment (and others). Some downgrading technique is marshaling configurations and load the configuration files to a cluster of lower version. currently even if we don't set `Extractors`, it will still generate extractors fields which is unrecognized by lower version DM.

Reply:
> Does this cause compatibility problems?

Yes. I expect `Extractors` should be totally hidden. Currently a task config exported by this version will cause "field Extractor not found in type router.TableRule" when loaded to old version.

Reply:
> I expect Extractors should be totally hidden

Seems `Extractors` can't be convert to pointer because it has `inline`, it will always initialized even if I set `,inline,omitempty`.

So, in this case, I think I have to delete the 'Extractors' field. and move 'TableExtractor/SchemaExtractor/SourceExtractor' to 'TableRule'. 

Or do you have any good ideas?

Reply:
It's good to me that we move `TableExtractor/SchemaExtractor/SourceExtractor` to `TableRule` 👍